geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradePartitionRegionClearServerVersionMismatch.java

```diff
@@ -109,20 +111,28 @@ public void before() {
       region.put("A", "ValueA");
       region.put("B", "ValueB");
     });
+  }
+
+  @After
+  public void after() {
+    locator.stop();
+    serverNew.stop();
+    serverOld.stop();
 
   }
 
   /**
-   * testClient_ServerVersionMismatchException - validates that when a client invokes a partitioned
-   * region clear on a cluster where one server is running an unsupported version for this feature
-   * we return a ServerVersionMismatchException
+   * testClient_ServerOperationException - validates that when a client invokes a partitioned region
+   * clear on a cluster where one server is running an unsupported version for this feature we
+   * return a ServerOperationException
    */
   @Test
-  public void testClient_ServerVersionMismatchException() throws Exception {
+  public void testClient_ServerOperationExceptionCurrentServerVersion() throws Exception {
     IgnoredException.addIgnoredException(ServerOperationException.class);
-    final int locatorPort = locator.getPort();
+
     // Get a client VM
-    ClientVM clientVM = cluster.startClientVM(3, c -> c.withLocatorConnection(locatorPort));
+    int serverPort = serverNew.getPort();
+    clientVM = cluster.startClientVM(3, oldVersion, c -> c.withServerConnection(serverPort));
```

Comment:
it's better to have one test to run pr clear from an old version and another test to run pr clear from a new version. Two test methods.

Reply:
Not sure I follow here, I have it broken out into two different client side tests. client -> old server, client -> new server

Reply:
I think Gester is recommending expanding the testing to include:

1. new client --> new server
2. new client --> old server
3. old client --> new server
4. old client --> old server

If (3) and (4) throw UnsupportedOperationException without sending any message to the server, then they could probably be collapsed into a single test.